main.tf

```diff
@@ -258,7 +275,7 @@ resource "aws_codedeploy_deployment_group" "default" {
   }
 
   dynamic "trigger_configuration" {
-    for_each = local.sns_topic_arn == null ? [0] : [1]
+    for_each = length(try(local.sns_topic_arn, "")) <= 0 ? [] : [0]
```

Comment:
- Why is this changed to `[0]` instead of `[1]` ?
- Why change the `== null` ?

Reply:
1. Because earlier [0] it was in the first place :D. But in fact, the value [1] will be more telling.
2. Previously, var.sns_topic_arn was set to null and module.sns_topic.sns_topic.arn could return an empty string. While this is logically impossible, I wanted to set the same types for those in the variables from which the local.sns_topic_arn ultimately is computed.